lib/sycamore/sycamore/tests/unit/transforms/test_similarity.py

```diff
@@ -0,0 +1,98 @@
+import ray
+
+from sycamore.data import Document, MetadataDocument
+from sycamore.plan_nodes import Node
+from sycamore.transforms.similarity import HuggingFaceTransformersSimilarityScorer, ScoreSimilarity
+
+
+class TestSimilarityScorer:
+
+    def test_transformers_similarity_scorer(self):
+        similarity_scorer = HuggingFaceTransformersSimilarityScorer()
+        score_property_name = "similarity_score"
+        query = "this is a cat"
+
+        dicts = [
+            {
+                "doc_id": 1,
+                "elements": [
+                    {"text_representation": "here is an animal with 4 legs and whiskers"},
+                ],
+            },
+            {
+                "doc_id": 2,
+                "elements": [
+                    {"id": 7, "text_representation": "this is a cat"},
+                    {"id": 1, "text_representation": "this is a dog"},
+                ],
+            },
+            {
+                "doc_id": 3,
+                "elements": [
+                    {"text_representation": "this is a dog"},
+                ],
+            },
+            {"doc_id": 4, "elements": [{"text_representation": "the number of pages in this document are 253"}]},
+            {  # handle empty element
+                "doc_id": 5,
+                "elements": [
+                    {"id": 1},
+                ],
+            },
+        ]
+        docs = [Document(item) for item in dicts]
+        result = similarity_scorer.generate_similarity_scores(
+            docs, query=query, score_property_name=score_property_name
+        )
+        result.sort(key=lambda doc: doc.properties.get(score_property_name, float("-inf")), reverse=True)
+        assert [doc.doc_id for doc in result] == [2, 1, 3, 4, 5]
+
+        assert result[0].properties[score_property_name + "_source_element_id"] == 7
+
+    def test_transformers_similarity_scorer_no_doc_structure(self):
+        similarity_scorer = HuggingFaceTransformersSimilarityScorer(ignore_doc_structure=True)
```

Comment:
maybe use a small model for unit/integ tests? perhaps cross-encoder/ms-marco-TinyBERT-L-2

Reply:
Ack

Reply:
Ended up using cross-encoder/ms-marco-MiniLM-L-2-v2 because the TinyBERT results were bad and felt wrong to use